nightfall-client/src/services/finalise-withdrawal.mjs

```diff
@@ -16,7 +16,9 @@ export async function finaliseWithdrawal(transactionHash) {
   const transactions = await Promise.all(
     block.transactionHashes.map(t => getTransactionByTransactionHash(t)),
   );
-  const index = transactions.findIndex(f => f.transactionHash === transactionHash);
+
+  // block can contain multiple transactions not in our database
+  const index = transactions.findIndex(f => transactionHash === f?.transactionHash ?? '0x0');
```

Comment:
Is there particular need for the nullish coalescer `?? '0x0'` ?  If `f` is `null` because it wasn't in the database then `f?.transactionHash` will be `undefined` and hence the test function `transactionHash === f?.transactionHash` will return false anyway. It's not harmful in anycase so happy to approve but just curious.

Reply:
Accessing attributes of `null` causes a `TypeError`.